### PR TITLE
ci: Setup code-coverage reporting (no-changelog)

### PR DESCRIPTION
**Pull request by netroy**

### CODE REVIEW

File: packages/cli/test/integration/commands/reset.cmd.test.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ afterAll(async () => {
 	await testDb.terminate();
 });
 
-test('user-management:reset should reset DB to default user state', async () => {
```
Comment (netroy) [inline packages/cli/test/integration/commands/reset.cmd.test.ts:29]: This test times out a lot more as the test suite grows. disabling it for now. Will re-enable it in https://github.com/n8n-io/n8n/pull/5464

File: packages/nodes-base/test/nodes/Helpers.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { readFileSync, readdirSync, mkdtempSync } from 'fs';
-import { BinaryDataManager, Credentials, loadClassInIsolation } from 'n8n-core';
```
Comment (netroy) [inline packages/nodes-base/test/nodes/Helpers.ts:2]: `loadClassInIsolation` loads the compiled filed from `dist`, in `vm` context, which prevents jest from instrumenting those files for coverage. That's why for test, I've switched this over to using `require` on the source typescript files instead.

File: jest.config.js [hunk 3]
```diff
@@ -22,11 +22,15 @@ const config = {
 	moduleNameMapper: {
 		'^@/(.*)$': '<rootDir>/src/$1',
 	},
+	collectCoverage: true,
+	coverageReporters: ['text'],
```
Comment (maspio) [inline jest.config.js:26]: I would prefer `coverageReporters: ['text-summary']` to not pollute the console logs too much. What do you think?

Comment (netroy) [reply jest.config.js:26]: how about using `text-summary` as the default, but if env variable `COVERAGE_REPORT` is set to `true`, we use `text` ?

Comment (maspio) [reply jest.config.js:26]: Yeah that sounds good to me 👍Easy enough

**Review by maspio — APPROVED** [review]

File: packages/cli/test/integration/commands/reset.cmd.test.ts (hunk 1 above)
Comment (despairblue) [reply packages/cli/test/integration/commands/reset.cmd.test.ts:29]: I re-enabled it: https://github.com/n8n-io/n8n/pull/9205

It wasn't timing out for me. Let me know if I should not re-enable it.

**Pull request merged**